quickshop-bukkit/src/main/java/com/ghostchu/quickshop/util/Util.java

```diff
@@ -497,6 +497,86 @@ public static boolean isEmptyComponent(@Nullable Component component) {
         return PlainTextComponentSerializer.plainText().serialize(component).isBlank();
     }
 
+    /**
+     * Find a string in a component
+     *
+     * @param component The component to check
+     * @param find The string to check for
+     * @return A boolean of whether the component contains the string
+     */
+    @NotNull
+    public static boolean findStringInComponent(@NotNull Component component, @NotNull String find) {
+        String plainText = PlainTextComponentSerializer.plainText().serialize(component).toLowerCase();
+        return plainText.replace(' ', '_').contains(find.toLowerCase());
+    }
+
+    /**
+     * Check for a string in a List of Components
+     *
+     * @param components A List<Component> of the components to check
+     * @param find The string to look for amongst the components
+     * @return A boolean of whether the string was found in the list of components
+     */
+    @NotNull
+    public static boolean findStringInList(@NotNull List<Component> components, @NotNull String find) {
+        for (Component name : components) {
+            if (findStringInComponent(name, find))
+                return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Get all enchants that can be found on an item stack
+     *
+     * @param itemStack The enchanted item
+     * @return The names of enchants contained on the enchanted item with levels
+     */
+    @NotNull
+    public static List<Component> getEnchantsForItemStack(@NotNull ItemStack itemStack) {
+        List<Component> enchants = new ArrayList<>();
+        if (!itemStack.hasItemMeta()) {
+            return enchants;
+        }
+
+        ItemMeta meta = itemStack.getItemMeta();
+        if (meta instanceof EnchantmentStorageMeta enchantmentStorageMeta && enchantmentStorageMeta.hasStoredEnchants()) {
+            for (Map.Entry<Enchantment, Integer> entry : enchantmentStorageMeta.getStoredEnchants().entrySet()) {
+                Component name = enchantmentDataToComponent(entry.getKey(), entry.getValue());
+                enchants.add(name);
+            }
+        } else {
+            for (Map.Entry<Enchantment, Integer> entry : meta.getEnchants().entrySet()) {
+                Component name = enchantmentDataToComponent(entry.getKey(), entry.getValue());
```

Comment:
Same thing here, a few things can be noted as final just for readability.
- enchants list
- component name
- ItemMeta